### PR TITLE
Version 2, with standalone text pre-processing

### PR DESCRIPTION
**Pull request by rspeer**
This version of wordfreq makes some breaking changes. Some of the exported functions have changed because we're separating text preprocessing, tokenization, and post-processing now.

The biggest change that will affect code is that "smash_numbers" is no longer an option to "tokenize" -- you do that with `lossy_tokenize`, which does some post-processing.

- Text preprocessing is a separate step from tokenization
- Optional post-processing of tokens to fit a wordlist happens in `lossy_tokenize`
- Wordlists are 'small', 'large', and the default 'best'
- The 'twitter' wordlist is gone, and 'combined' is a deprecated alias for 'small'
- Chinese tokenization doesn't smash its output into Simplified Chinese, though Simplified Chinese is still used for lookups
- Decisions about how to handle each language are all in one place, the `language_info` module
- Tokenizing CJK languages with punctuation involves a less confusing chain of events
- Preprocessing includes transliteration for Azerbaijani, because we can. (This is for the benefit of ConceptNet; we don't have an Azerbaijani word frequency list.)
- New data from `exquisite_corpus`

### CODE REVIEW

File: wordfreq/__init__.py [hunk 1]
```diff
@@ -30,8 +32,9 @@
 # frequency.)
 INFERRED_SPACE_FACTOR = 10.0
 
-# simple_tokenize is imported so that other things can import it from here.
-# Suppress the pyflakes warning.
+# tokenize and simple_tokenize are imported so that other things can import
+# them from here. Suppress the pyflakes warning.
+tokenize = tokenize
 simple_tokenize = simple_tokenize
```
Comment (alin-luminoso) [inline wordfreq/__init__.py]: Drive-by: it turns out pyflakes is just as satisfied if you do
```
tokenize
simple_tokenize
```

File: wordfreq/tokens.py [hunk 2]
```diff
@@ -364,39 +186,66 @@ def tokenize(text, lang, include_punctuation=False, external_wordlist=False,
...
+        from wordfreq.chinese import jieba_tokenize as _jieba_tokenize
+        tokens = _jieba_tokenize(text, external_wordlist=external_wordlist)
+        if not include_punctuation:
+            tokens = [token for token in tokens if not PUNCT_RE.match(token)]
+    else:
+        # This is the default case where we use the regex tokenizer. First
+        # let's complain a bit if we ended up here because we don't have an
+        # appropriate tokenizer.
+        if info['tokenizer'] != 'regex':
+            if lang not in _WARNED_LANGUAGES:
```
Comment (Tahnan) [inline wordfreq/tokens.py]: Nitpick: worth simplifying to `if info['tokenizer'] != 'regex' and lang not in _WARNED_LANGUAGES`?

File: wordfreq/language_info.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    'Khmr',  # Khmer script
+    'Laoo',  # Lao script
+    'Mymr',  # Burmese script
+    'Tale',  # Tai Le script
+    'Talu',  # Tai Lü script
+    'Lana',  # Lanna script
+]
+
+
+def _language_in_list(language, targets, min_score=80):
```
Comment (Tahnan) [inline wordfreq/language_info.py]: `min_score` isn't used--not sure if it should be removed or if it should replace the `0` in line 35.

**Comment by Tahnan** [comment]
...one more thing, while I'm thinking about it, as long as several changes are going in.

If you ask for a wordlist that doesn't exist in a language that does, `get_frequency_list` raises the error `LookupError("No wordlist available for language %r" % lang)`.  That did confuse me, when I was trying to get a long list for a language--it turned out that there was no long list available, only short, but the error message made me think I had the wrong language code.  Do you think it makers sense to change this to `LookupError("No wordlist %s available for language %r" % (wordlist, lang))`?

(At that time, I discovered I wanted `available_wordlists(lang)` as a kind of inverse of `available_languages(wordlist)`.  With the removal of "twitter", though, and the ability to just use "best" rather than having to figure out if the language's wordlist is "long" or "short", that's probably less necessary than it was.)
